### PR TITLE
Change focus_on_window_activation var to allow functions

### PR DESCRIPTION
**Pull request by MoonCubes**
When the _NET_ACTIVE_WINDOW message is send the focus_on_window_activation config variable is checked.
If it is a function (that takes the window as an argument) and returns True the window is focused. If it is a string then the old behaviors are used.
This could for example be used if you only want to focus the browser when you open a link, and it's in a different group.

This is how you could set that up:
```py
def focus_behavior(client: Window):
    return "Firefox-esr" in client.get_wm_class() or client.group.screen == qtile.current_screen
focus_on_window_activation = focus_behavior
```

### CODE REVIEW

File: docs/manual/commands/keybindings.rst [hunk 1]
```diff
@@ -11,6 +11,7 @@ Default configuration
 .. LS_PNG
 .. image:: /_static/keybindings/mod4.png
 .. image:: /_static/keybindings/mod4-shift.png
+.. image:: /_static/keybindings/control-mod1.png
```
Comment (tych0) [inline docs/manual/commands/keybindings.rst:14]: This seems like a separate fix, right? can you put it in a separate patch?

File: docs/manual/config/index.rst [hunk 2]
```diff
@@ -127,6 +127,9 @@ configuration variables that control specific aspects of Qtile's behavior:
         - smart: automatically focus if the window is in the current group
 
         - never: never automatically focus any window that requests it
+    * - ``window_activation_rules``
+      - ``[]``
+      - A list of Match objects of windows that will always be focused when the _NET_ACTIVE_WINDOW message is sent by applications.
```
Comment (tych0) [inline docs/manual/config/index.rst:132]: I wonder: can we make `focus_on_window_activation` take a function, so that people can define this stuff on their own?

Comment (MoonCubes) [reply docs/manual/config/index.rst:132]: @tych0 Should `focus_on_window_activation` only take a function or either a function or a string with the old behaviors? (so old configs don't break)

Comment (tych0) [reply docs/manual/config/index.rst:132]: Yeah, we'll want old configs to stay working, so I think a string or a function should be allowable.

**Comment by tych0** [comment]
Thanks, looks reasonable to me. Can you add a test to make sure we don't break this?

**Comment by MoonCubes** [comment]
Done. I added it to the test_urgent_hook_fire test
